src/controller/CHIPDeviceController.cpp

```diff
@@ -908,6 +920,7 @@ CHIP_ERROR DeviceCommissioner::Commission(NodeId remoteDeviceId)
     mDefaultCommissioner->SetOperationalCredentialsDelegate(mOperationalCredentialsDelegate);
     if (device->IsSecureConnected())
     {
+        MATTER_LOG_METRIC_BEGIN(kMetricDeviceCommissionerCommission);
```

Comment:
So is kMetricDeviceCommissionerCommission a metric that gets begin/end?  How do the `ReturnErrorOnFailureWithMetric`/`VerifyOrDoWithMetric` bits interact with begin/end?

Reply:
The ReturnErrorOnFailureWithMetric/VerifyOrDoWithMetric are used to handle any failure that happen synchronously before the log async operation to start commissioning.

Reply:
Yup